Mathlib/MeasureTheory/Function/UnifTight.lean

```diff
@@ -0,0 +1,372 @@
+/- Lemma used in `tendsto_Lp_notFinite_of_tendsto_ae`.
+   XXX: Alternative name: `ae_tendsto_ae_congr`? -/
+theorem tendsto_ae_congr_ae {f f' : ℕ → α → β} {g g' : α → β}
+    (hff' : ∀ (n : ℕ), f n =ᵐ[μ] f' n) (hgg' : g =ᵐ[μ] g')
+    (hfg : ∀ᵐ x ∂μ, Tendsto (fun n => f n x) atTop (𝓝 (g x))) :
+    ∀ᵐ x ∂μ, Tendsto (fun n => f' n x) atTop (𝓝 (g' x)) := by
+  have hff'' := eventually_countable_forall.mpr hff'
+  filter_upwards [hff'', hgg', hfg] with x hff'x hgg'x hfgx
+  apply Tendsto.congr hff'x
+  rw [← hgg'x]; exact hfgx
+
+theorem tendsto_Lp_notFinite_of_tendsto_ae (hp : 1 ≤ p) (hp' : p ≠ ∞)
+    {f : ℕ → α → β} {g : α → β} (haef : ∀ n, AEStronglyMeasurable (f n) μ)
+    (hg' : Memℒp g p μ) (hui : UnifIntegrable f p μ) (hut : UnifTight f p μ)
+    (hfg : ∀ᵐ x ∂μ, Tendsto (fun n => f n x) atTop (𝓝 (g x))) :
+    Tendsto (fun n => snorm (f n - g) p μ) atTop (𝓝 0) := by
+  -- come up with an a.e. equal strongly measurable replacement `f` for `g`
+  have hf := fun n => (haef n).stronglyMeasurable_mk
+  have hff' := fun n => (haef n).ae_eq_mk (μ := μ)
+  have hui' := hui.ae_eq hff'
+  have hut' := hut.ae_eq hff'
+  have hg := hg'.aestronglyMeasurable.stronglyMeasurable_mk
+  have hgg' := hg'.aestronglyMeasurable.ae_eq_mk (μ := μ)
+  have hg'' := hg'.ae_eq hgg'
+  have haefg' := tendsto_ae_congr_ae hff' hgg' hfg
+  set f' := fun n => (haef n).mk (μ := μ)
+  set g' := hg'.aestronglyMeasurable.mk (μ := μ)
+  have haefg (n : ℕ) : f n - g =ᵐ[μ] f' n - g' := (hff' n).sub hgg'
+  have hsnfg (n : ℕ) := snorm_congr_ae (p := p) (haefg n)
+  apply Filter.Tendsto.congr (fun n => (hsnfg n).symm)
+  exact tendsto_Lp_notFinite_of_tendsto_ae_of_meas hp hp' hf hg hg'' hui' hut' haefg'
+
+
+/-- Forward direction of Vitali's convergence theorem (non-finite version):
+    if `f` is a sequence of uniformly integrable, uniformly tight functions that converge in
+    measure to some function `g` in a finite measure space, then `f` converge in Lp to `g`. -/
+theorem tendsto_Lp_notFinite_of_tendstoInMeasure (hp : 1 ≤ p) (hp' : p ≠ ∞)
+    (hf : ∀ n, AEStronglyMeasurable (f n) μ) (hg : Memℒp g p μ)
+    (hui : UnifIntegrable f p μ) (hut : UnifTight f p μ)
+    (hfg : TendstoInMeasure μ f atTop g) : Tendsto (fun n => snorm (f n - g) p μ) atTop (𝓝 0) := by
+  refine tendsto_of_subseq_tendsto fun ns hns => ?_
+  obtain ⟨ms, _, hms'⟩ := TendstoInMeasure.exists_seq_tendsto_ae fun ε hε => (hfg ε hε).comp hns
+  exact ⟨ms,
+    tendsto_Lp_notFinite_of_tendsto_ae hp hp' (fun _ => hf _) hg
+      (fun ε hε => -- `UnifIntegrable` on a subsequence
+        let ⟨δ, hδ, hδ'⟩ := hui hε
+        ⟨δ, hδ, fun i s hs hμs => hδ' _ s hs hμs⟩)
+      (fun ε hε => -- `UnifTight` on a subsequence
+        let ⟨s, hμs, hfε⟩ := hut hε
+        ⟨s, hμs, fun i => hfε _⟩)
+      hms'⟩
+
+
+/-- **Vitali's convergence theorem** (non-finite measure version):
+    A sequence of functions `f` converges to `g` in Lp
+    if and only if it is uniformly integrable, uniformly tight and to `g` in measure. -/
```

Comment:
```suggestion
/-- **Vitali's convergence theorem** (non-finite measure version).

A sequence of functions `f` converges to `g` in Lp
if and only if it is uniformly integrable, uniformly tight and to `g` in measure. -/
```
to follow the short docstring-long docstring style.

Reply:
OK. A similar update could be done in UniformIntegrable.lean.